metrics/metricinfo.go

```diff
@@ -96,6 +112,85 @@ func (m *MetricInfo) collectOnce(ctx context.Context) error {
 	return nil
 }
 
+func (m *MetricInfo) collectOnceFromFile(t *testing.T, fileName string) {
```

Comment:
Can these functions simply return an error and then the caller can check for errors?

Reply:
Refactored it like that.